crates/wasi-http/src/streams_impl.rs

```diff
@@ -111,7 +140,11 @@ impl crate::streams::Host for WasiHttp {
     }
 
     fn drop_output_stream(&mut self, stream: OutputStream) -> wasmtime::Result<()> {
-        self.streams.remove(&stream);
+        let st = self
+            .streams
+            .get_mut(&stream)
+            .ok_or_else(|| anyhow!("stream not found: {stream}"))?;
```

Comment:
Doesn't `with_context` work here?

Reply:
I'm not sure I'm following your suggestion, can you clarify?

Reply:
Sure, I'm referring to this method on the `anyhow::Context` trait https://docs.rs/anyhow/latest/anyhow/trait.Context.html#tymethod.with_context

If it's in scope, you should be able to call `context` or `with_context` (usually used with `format!` and similar) on most `Result` and all `Option` types directly, thus eliminating the need for `ok_or_else` and explicit `anyhow!` invocation

Reply:
I made those ok_or_else(|| anyhow!(...)) suggestions in a prior PR and wasn't aware that Context could do that for me. Thank you.

@brendandburns The suggested change here is `let st = self.streams.get_mut(&stream).with_context(|| format!("stream not found: {stream}"))?;`, or maybe even `....get_mut(&stream).context("stream not found")?` if leaving out the stream number is ok - and it probably is, because (at least when using the component linker, I'm not sure if it works in module mode) bindgen is inserting `tracing` statements which someone debugging can recover the values of the arguments from.